src/toolkits/activity_classification/ac_data_iterator.cpp

```diff
@@ -425,39 +566,70 @@ data_iterator::batch simple_data_iterator::next_batch(size_t batch_size) {
     labels.resize(labels_size, 0.f);
     weights.resize(labels_size, 0.f);
   }
+  std::vector<batch::chunk_info> batch_info;
+  batch_info.reserve(batch_size);
 
   // Iterate through SFrame rows until filling the batch or reaching the end of
   // the data.
   float* features_out = features.data();
   float* labels_out = labels.data();
   float* weights_out = weights.data();
-  std::vector<batch::chunk_info> batch_info;
-  batch_info.reserve(batch_size);
+
   while (batch_info.size() < batch_size && next_row_ != end_of_rows_) {
 
     const sframe_rows::row& row = *next_row_;
 
+    const flex_int &chunk_length = row[chunk_len_column_index].get<flex_int>();
```

Comment:
Nit: there’s no reason to store a const reference to a `flex_int`. Unlike an object, it’s cheaper to copy than to use a reference